camayoc/qcs_models.py

```diff
@@ -287,3 +291,127 @@ def equivalent(self, other):
                 if not other.get(key) == value:
                     diffs += 1
         return diffs == 0
+
+
+class Scan(QCSObject):
+    """A class to aid in CRUD test cases for scan jobs.
+
+    Scan jobs can be created on the quipucords server by instantiating a Scan
+    object. The id of an existing NetworkProfile is necessary to create a scan
+    job.
+
+    Example::
+        >>> from camayoc.qcs_models import NetworkProfile, HostCredential, Scan
+        >>>
+        >>> hostcred = HostCredential(password='foo')
+        >>> hostcred.create()
+        >>> netprof = NetworkProfile(hosts=['0.0.0.0'],
+                                     credential_ids=[hostcred._id])
+        >>> netprof.create()
+        >>> scan = Scan(profile_id=netprof._id)
+        >>> scan.create()
+        >>> scan.pause()
+        >>> assert scan.status() == 'paused'
+    """
+
+    def __init__(
+            self,
+            client=None,
+            profile_id=None,
+            max_concurrency=50,
+            scan_type='host',
+            _id=None):
+        """Iniitalize a Scan object with given data.
+
+        If no value for max_concurrency is given, it will be set to 50, and if
+        no value for scan_type is given, the default type is host (for which
+        facts are collected). The other valid option for scan_type is
+        'discovery' in which no facts are collected, the server simply sees how
+        many hosts in the profile it can make contact and log into.
+        """
+        super().__init__(client=client, name=None, _id=_id)
+
+        # scan jobs do not have a name, so we delete this attribute
+        del self.name
+        self.profile = profile_id
+        self.endpoint = QCS_SCAN_PATH
+
+        # valid scan types are 'host' and 'discovery'
+        self.scan_type = scan_type
+        self.max_concurrency = max_concurrency
+
+    def delete(self):
+        """No delete method is implemented for scan objects.
+
+        Emit a warning to the camayoc user instead of doing anything.
+        """
+        LOGGER.critical(
+            'Camayoc Warning: The DELETE method is not allowed for scans')
```

Comment:
What do you think about this? I could instead raise an exception, the main thing is that scans don't take the delete method so you would not want to pass scan objects to the cleanup fixture.